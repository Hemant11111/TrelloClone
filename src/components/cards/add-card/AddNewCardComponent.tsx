import React, { useCallback } from "react";
import { useDispatch } from "react-redux";

import "./AddNewCard.scss";
import { createDummyNewCardThunk } from "../../../redux/thunk/card.thunk";

interface AddNewCardComponentProps {
}


export default function AddNewCardComponent(props: AddNewCardComponentProps) {

    const dispatch = useDispatch();

    const handleAddCardClick = useCallback(() => {
        dispatch(createDummyNewCardThunk());
    }, []);

    return (
        <div className="add-new-card-container">
            <button data-testid="add-new-card-button" className="add-new-card-btn" onClick={handleAddCardClick}>
                <i className="fa fa-plus"/>
            </button>
        </div>
    )
}