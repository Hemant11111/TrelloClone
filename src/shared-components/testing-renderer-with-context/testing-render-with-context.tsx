import { ReactElement } from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import store from "../../redux/store/store";

export function testingRenderWithContext(element: ReactElement) {
    return render(
        <Provider store={store}>
            {element}
        </Provider>
    )
}