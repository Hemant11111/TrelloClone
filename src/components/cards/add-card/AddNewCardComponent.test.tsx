import React from 'react';
import { screen } from '@testing-library/react';

import AddNewCardComponent from "./AddNewCardComponent";
import { testingRenderWithContext } from "../../../shared-components/testing-renderer-with-context/testing-render-with-context";


describe('AddNewCardComponent', () => {

    it('should match snapshot', () => {
        const {asFragment} = testingRenderWithContext(<AddNewCardComponent/>);
        expect(asFragment()).toMatchSnapshot();
    })


    it('should render Add button', () => {
        testingRenderWithContext(<AddNewCardComponent/>);
        expect(screen.getByTestId("add-new-card-button")).toBeInTheDocument();
    })
})