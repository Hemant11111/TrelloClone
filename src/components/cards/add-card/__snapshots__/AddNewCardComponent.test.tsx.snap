// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewCardComponent should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="add-new-card-container"
  >
    <button
      class="add-new-card-btn"
      data-testid="add-new-card-button"
    >
      <i
        class="fa fa-plus"
      />
    </button>
  </div>
</DocumentFragment>
`;
